### PR TITLE
Update input event tutorial to reflect recent changes

### LINKED ISSUE #2581: _unhandled_key_input method missing from documentation
In the documentation stating flow order for input events (https://docs.godotengine.org/en/3.1/tutorials/inputs/inputevent.html#what-is-it)

_unhandled_key_input()

Is missing from the diagram and all text referring to input events other than the class library documentation.

It is called after _unhandled_input(), but I am not sure if it is called before or after CollisionObject _input_event()

Hope this helps.

Current as of 3.1 documentation

### LINKED ISSUE #2619: The InputEvent graph is outdated
http://docs.godotengine.org/en/3.1/tutorials/inputs/inputevent.html

It mentions `_input_event` for GUI, but now it should be `_gui_input`.

### LINKED ISSUE #3369: Outdated "How does it work" tutorial about Input Events
The tutorial "How does it work" about Input Events
https://docs.godotengine.org/en/stable/tutorials/inputs/inputevent.html#how-does-it-work
diverges in many details from the current implementation.

I suggest to add a warning to this chapter, so that inexperienced users are made aware of the fact that not all information within this chapter is up to date.

I am aware of the following aspects, where the tutorial diverges from the actual implementation:
- https://github.com/godotengine/godot/issues/25525
- #2581
- #2619
- viewports and their children usually never receive input events
- in some cases parents receive input events before their children (ViewportContainer related)
- in some cases the root node never receives an _gui_input event, although all Control-children have their mouse_filter set to "pass" (ViewportContainer related)
- in some cases a key triggers an _unhandled_key_input() event for a node and afterwards triggers an _unhandled_input() event for a different node (ViewportContainer related)

As an example, here is an actual order of _input key-events (Godot V3.2.1):

_input_Order

### PR DESCRIPTION
**Pull request by Sauermann**
This PR updates the InputEvent tutorial

- resolve #2581
- resolve #2619
- resolve #3369

Update input event flow graphic to reflect the changes of
- https://github.com/godotengine/godot/pull/57894
- https://github.com/godotengine/godot/pull/79248
Made some descriptions more precise.

Input Event table:
- Added missing InputEvents Gesture, MIDI and Shortcut
- Removed unused Type Index column

Updated 2023-08-05: include 4.2-changes

### CODE REVIEW

**Comment by Piralein** [comment]
Note for merge: the upstream PR is not marked for cherrypick-4.0 and has to be considered 4.1 documentation.

**Comment by mhilbrunner** [comment]
@Sauermann Seeing the first PR got merged and included in 4.1, and the second one at this point will be 4.2+, does it make sense to merge this and cherry-pick to 4.1, then do another one if further changes are needed for 4.2+?

**Comment by Sauermann** [comment]
@mhilbrunner That idea makes sense with respect to the content of the article.
However https://github.com/godotengine/godot/pull/79248 will make it necessary to update the graphic again, which was my reason for waiting in order to keep the git-history small.

**Comment by Sauermann** [comment]
The other PR has been in circulation for two weeks now and got some testing. So I believe, that it is unlikely to be reverted.
I have updated the PR with the necessary changes for that PR.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
We can wait a bit more then merge. LGTM.

File: tutorials/inputs/inputevent.rst [hunk 1]
```diff
@@ -152,37 +152,47 @@ anything and only contains some basic information, such as event ID
...
+| :ref:`InputEvent <class_InputEvent>`                              | Empty Input Event.                      |
++-------------------------------------------------------------------+-----------------------------------------+
+| :ref:`InputEventKey <class_InputEventKey>`                        | Contains a keycode and Unicode value,   |
+|                                                                   | as well as modifiers.                   |
++-------------------------------------------------------------------+-----------------------------------------+
+| :ref:`InputEventMouseButton <class_InputEventMouseButton>`        | Contains click information, such as     |
+|                                                                   | button, modifiers, etc.                 |
++-------------------------------------------------------------------+-----------------------------------------+
+| :ref:`InputEventMouseMotion <class_InputEventMouseMotion>`        | Contains motion information, such as    |
+|                                                                   | relative, absolute positions and speed. |
```
Comment (AThousandShips) [inline tutorials/inputs/inputevent.rst:167]:
```suggestion
|                                                                   | relative and absolute positions and speed. |
```

Comment (Sauermann) [reply tutorials/inputs/inputevent.rst]: Thanks, I have included this change.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
@Sauermann Merged, thanks for all this work! I guess for 4.1 a separate PR is easiest, backporting the relevant parts?

**Comment by Sauermann** [comment]
Yes, that sounds about right.

**Pull request merged**